### PR TITLE
fix(widgets): widget style prop keys should accept camelCase css properties and dashed css variables

### LINKED ISSUE #8990: [Bug] Widget `style` prop does not apply styles for camel cased style keys
### Description

When I pass a `style` object to widgets, styles for camel cased attributes (`borderColor`, `zIndex`, etc) aren't applied. Setting styles with dash cased keys does work but that violates the `CSSStyleDeclaration` type.

### Flavors

- [ ] Script tag
- [ ] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

I expect to see all style applied to the widget while still complying with the typings

### Steps to Reproduce

Here is a link to a branch of a repo that can be used to reproduce this issue: https://github.com/TylerMatteo/react-deck-boilerplate/tree/widget-dashcase-style-bug

To reproduce:
1. Clone this repo and check out the `widget-dashcase-style-bug` branch
2. `npm install`
3. `npm run dev`
4. Observe that the widget styles applied to `ZoomWidget` in `/src/App.tsx` aren't applied.

### Environment

- Framework version: deck.gl@9.0.20
- Browser: Firefox 125.0.3
- OS: MacOS Ventura 13.5

### Logs

N/A

### PR DESCRIPTION
**Pull request by TylerMatteo**
Closes #8990 

#### Change List
- Updated the  `style` prop on `CompassWidget`, `ZoomWidget`, and `FullscreenWidget` to support camelCase CSS properties (e.g. `backgroundColor`) and kabab-case CSS variables (e.g. `--button-size`).

### CODE REVIEW

**Comment by Pessimistress** [comment]
@chrisgervang if I remember right, we used to use `Object.assign(element.style, style)` which only worked for camel case property names. It was switched to `element.style.setProperty` in order to support CSS variables.

Alternatively we could just do this the dirty way by calling both of the above...

**Comment by chrisgervang** [comment]
@Pessimistress, that's correct, but I didn't account for camelCase styles. It's probably best to have a solution that supports at least variables and camelCase, or all three (even if kebab-case isn't popular).

I think there are three solutions to consider.

1. Simple to understand and supports variables and camelCase:
```js
if (property.startsWith('--')) {
  // Assume CSS variable
  element.style.setProperty(property, value);
} else {
  // Assume camelCase
  element.style[property] = value;
}
```

2. The next being suggested by this PR. @TylerMatteo's solution works with all three input types. It's pretty good, but asking every widget author to understand regex isn't my preference.

3. Being explicit with handling each input.

```js
if (property.startsWith('--')) {
  // Assume CSS variable
  element.style.setProperty(property, value);
} else if (property.includes('-')) {
  // Convert kebab-case to camelCase
  const camelCaseProperty = property.replace(/-([a-z])/g, (match, letter) => letter.toUpperCase());
  element.style[camelCaseProperty] = value;
} else {
  // Assume camelCase
  element.style[property] = value;
}
```

I've uploaded them into a [gist](https://gist.github.com/chrisgervang/4b33c14466c050da4bd0276ae45d4ed6) for easy testing.

----

While `element.style[property] = value;` sometimes works with kebab-case in Chrome, it's not a standard behavior we should rely on. I think I'd prefer to go with option 1 unless there's a strong argument to support camelCase and kebab-case in JS. Core widgets are meant to be examples in themselves, so I'd like to keep them simple to read and understand.

**Comment by TylerMatteo** [comment]
@chrisgervang sounds reasonable to me - I hadn't considered css variables. Shall I go ahead and update my branch to implement solution 1?

**Comment by chrisgervang** [comment]
Yes, thank you! Could you also please take a look at the `style` prop docs to see if they need clarification?

**Comment by Pessimistress** [comment]
It might be easier to create a shared function `applyStyles(element, styles)`.

**Comment by TylerMatteo** [comment]
@chrisgervang I made that change and created a shared `applyStyles` util added to `core`. Let me know if you'd prefer this change stay scoped to the `widgets` module.

File: test/modules/core/utils/apply-styles.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import {applyStyles} from '@deck.gl/core/utils/apply-styles';
+
+const TEST_CASES = [
+  {
+    title: 'CSS variable',
+    argument: {property: '--my-var', value: 'red'},
+    result: {property: '--my-var', value: 'red'}
+  },
+  {
+    title: 'camelCase property',
```
Comment (chrisgervang) [inline test/modules/core/utils/apply-styles.spec.ts:12]: Can you please test the faulty Kebab-case as well? I wonder what our runner will do

File: modules/widgets/src/fullscreen-widget.tsx [hunk 2]
```diff
@@ -99,13 +97,17 @@ export class FullscreenWidget implements Widget<FullscreenWidgetProps> {
...
-          );
+          Object.entries(oldProps.style).map(([key]) => {
+            if (key.startsWith('--')) {
+              // Assume CSS variable
+              el.style.removeProperty(key);
+            } else {
+              // Assume camelCase
+              el.style[key] = '';
+            }
+          });
```
Comment (chrisgervang) [inline modules/widgets/src/fullscreen-widget.tsx:108]: Let's create a removeStyles as well. This seems useful to apply in all widgets when style has changed

**Review by chrisgervang — COMMENTED** [review]
Great progress! Left some comments. @Pessimistress would be the one to make a call on whether she wants these utility functions in core or not. I think we could consider whether or not it'd be useful to use the Deck class too.

**Pull request merged**